core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -572,20 +578,28 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
   }
   private def createWdsAppInWorkspace(workspaceId: UUID,
                                       parentContext: RawlsRequestContext,
-                                      sourceWorkspaceId: Option[UUID]
-  ): Future[Unit] = {
-    // create a WDS application in Leo. Do not fail workspace creation if WDS creation fails.
-    logger.info(s"Creating WDS instance [workspaceId = ${workspaceId}]")
-    traceWithParent("createWDSInstance", parentContext)(_ =>
-      Future(
-        leonardoDAO.createWDSInstance(parentContext.userInfo.accessToken.token, workspaceId, sourceWorkspaceId)
-      )
-        .recover { case t: Throwable =>
-          // fail silently, but log the error
-          logger.error(s"Error creating WDS instance [workspaceId = ${workspaceId}]: ${t.getMessage}", t)
-        }
-    )
-  }
+                                      sourceWorkspaceId: Option[UUID],
+                                      workspaceAttributeMap: AttributeMap
+  ): Future[Unit] =
+    workspaceAttributeMap.get(AttributeName.withDefaultNS("disableAutomaticAppCreation")) match {
+      case Some(AttributeBoolean(true)) =>
+        // Skip WDS deployment for testing purposes.
+        logger.info("Skipping creation of WDS per request attributes")
+        Future.successful()
+      case _ =>
+        // create a WDS application in Leo. Do not fail workspace creation if WDS creation fails.
+        logger.info(s"Creating WDS instance [workspaceId = ${workspaceId}]")
+        Future.successful()
```

Comment:
Why is this `Future.successful` needed here?

Reply:
The method returns Future.unit. Is Future.successful() not the correct way to do this?

Reply:
Oops!! I see now what you are talking about-- let me remove that!

Reply:
@aherbst-broad fixed, and thanks for catching that!